src/test/java/teammates/test/cases/logic/CoursesLogicTest.java

```diff
@@ -438,12 +438,22 @@ private void testGetCoursesForStudentAccount() throws Exception {
         assertEquals(2, courseList.size());
 
         CourseAttributes course1 = dataBundle.courses.get("typicalCourse1");
-        assertEquals(course1.getId(), courseList.get(0).getId());
-        assertEquals(course1.getName(), courseList.get(0).getName());
 
         CourseAttributes course2 = dataBundle.courses.get("typicalCourse2");
-        assertEquals(course2.getId(), courseList.get(1).getId());
-        assertEquals(course2.getName(), courseList.get(1).getName());
+
+        boolean isCourse1Present = false;
+        boolean isCourse2Present = false;
+
+        for (CourseAttributes course : courseList) {
+            if (course.getId().equals(course1.getId()) && course.getName().equals(course1.getName())) {
+                isCourse1Present = true;
+            } else if (course.getId().equals(course2.getId()) && course.getName().equals(course2.getName())) {
+                isCourse2Present = true;
+            }
+        }
+
+        assertTrue(isCourse1Present);
+        assertTrue(isCourse2Present);
```

Comment:
Cannot we solve this like this by creating another List and of courses and sorting both the list by Id 

        List<CourseAttributes> courseList = coursesLogic
                .getCoursesForStudentAccount(studentInTwoCourses.googleId);
        CourseAttributes.sortById(courseList);
        assertEquals(2, courseList.size());
        
        CourseAttributes course1 = dataBundle.courses.get("typicalCourse1");

        CourseAttributes course2 = dataBundle.courses.get("typicalCourse2");

        List<CourseAttributes> courses= new  ArrayList<CourseAttributes>();
        courses.add(course1);
        courses.add(course2);
       
        CourseAttributes.sortById(courses);
       
        assertEquals(courses.get(0).getId(), courseList.get(0).getId());
        assertEquals(courses.get(0).getName(), courseList.get(0).getName());

        assertEquals(courses.get(1).getId(), courseList.get(1).getId());
        assertEquals(courses.get(1).getName(), courseList.get(1).getName());`